src/Symfony/Component/Security/Core/Tests/Authentication/Provider/DaoAuthenticationProviderTest.php

```diff
@@ -299,3 +292,30 @@ protected function getProvider($user = null, $userChecker = null, $passwordEncod
         return new DaoAuthenticationProvider($userProvider, $userChecker, 'key', $encoderFactory);
     }
 }
+
+class TestUser implements UserInterface
```

Comment:
💡 `UserInterface` wasn't mocked properly in the tests: `getPassword()` returned `null` where it shouldn't. This caused problems with the new type-hints, so I switched to a dummy implementation instead of bloating the mocking code.